runtime/src/main/java/org/corfudb/runtime/collections/CorfuStreamEntry.java

```diff
@@ -0,0 +1,60 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+import lombok.Getter;
+
+/**
+ * This will be the type of each entry returned by CorfuStore's Streamer Interface
+ *
+ * @param <K> - type of the protobuf KeySchema defined while the table was created.
+ * @param <V> - type of the protobuf PayloadSchema defined by the table creator.
+ * @param <M> - type of the protobuf metadata schema defined by table creation.
+ *
+ *  NOTE: Ensure that the above protobuf generated classes are accessible to your JVM!
+ *
+ *  Created by hisundar on 2019-10-18
+ */
+public class CorfuStreamEntry<K extends Message, V extends Message, M extends Message> {
+    /**
+     * Key of the UFO stream entry
+     */
+    @Getter
+    private final K key;
+
+    /**
+     * Value of the UFO stream entry
+     */
+    @Getter
+    private final V payload;
+
+    /**
+     * Metadata (ManagedResource) of the UFO stream entry
+     */
+    @Getter
+    private final M metadata;
+
+    /**
+     * Defines the type of the operation in this stream
+     */
+    public enum OperationType {
```

Comment:
hopefully there are no migration / upgrade implications if a new operation is added later on.

Reply:
We should be able to add/remove from this Type as long as we can fix up the consumers.